be/src/exec/pipeline/operator.h

```diff
@@ -134,6 +134,10 @@ class Operator {
     static const int32_t s_pseudo_plan_node_id_for_result_sink;
     static const int32_t s_pseudo_plan_node_id_upper_bound;
 
+    std::shared_ptr<RuntimeProfile>& runtime_profile() { return _runtime_profile; }
+    std::shared_ptr<RuntimeProfile>& common_metrics() { return _common_metrics; }
+    std::shared_ptr<RuntimeProfile>& unique_metrics() { return _unique_metrics; }
```

Comment:
Maybe raw pointer is better? Lifecycle of profile should be maintained by operator itself